Bio/SeqIO/AbiIO.py

```diff
@@ -376,7 +366,17 @@ def AbiIterator(source, alphabet=None, trim=False):
 
         if marker != b"ABIF":
             raise OSError("File should start ABIF, not %r" % marker)
-
+        records = self.iterate(handle)
+        return records
+
+    def iterate(self, handle):
```

Comment:
Again this has to do with timing. The `parse` function is called immediately when the parser is created. Also this function is not a generator; it will return immediately. This allows you to parse the file header, and to get information from the file header without having to start pulling out records:
```
>>> records = SeqIO.parse("dna_example.xml", "seqxml")
# Just the file header is parsed at this point
>>> records.source
'Ensembl'
>>> records.sourceVersion
'56'
```

If `parse` and `iterate` are combined into a single function, then this will be a generator, which is not called until we actually start pulling out records. Then the file header is also not parsed initially:
```
>>> records = SeqIO.parse("dna_example.xml", "seqxml")
# Nothing is parsed at this point
>>> records.source
# not defined
>>> record = next(records)
>>> records.source
'Ensembl'
```